src/simplewallet/simplewallet.cpp

```diff
@@ -6583,34 +6608,20 @@ bool simple_wallet::transfer_main(const std::vector<std::string> &args_, bool ca
     de.is_subaddress = info.is_subaddress;
     de.is_integrated = info.has_payment_id;
 
-    if (info.has_payment_id || !payment_id_uri.empty())
+    if (info.has_payment_id)
     {
       if (payment_id_seen)
       {
-        fail_msg_writer() << tr("a single transaction cannot use more than one payment id");
-        return false;
+          fail_msg_writer() << tr("a single transaction cannot use more than one payment id");
+          return false;
       }
 
-      crypto::hash payment_id;
       std::string extra_nonce;
-      if (info.has_payment_id)
-      {
-        set_encrypted_payment_id_to_tx_extra_nonce(extra_nonce, info.payment_id);
-      }
-      else if (tools::wallet2::parse_payment_id(payment_id_uri, payment_id))
-      {
-        LONG_PAYMENT_ID_SUPPORT_CHECK();
-      }
-      else
-      {
-        fail_msg_writer() << tr("failed to parse payment id, though it was detected");
-        return false;
-      }
-      bool r = add_extra_nonce_to_tx_extra(extra, extra_nonce);
-      if(!r)
+      set_encrypted_payment_id_to_tx_extra_nonce(extra_nonce, info.payment_id);
+      if (!add_extra_nonce_to_tx_extra(extra, extra_nonce))
       {
-        fail_msg_writer() << tr("failed to set up payment id, though it was decoded correctly");
-        return false;
+          fail_msg_writer() << tr("failed to set up payment id");
```

Comment:
Please check here and in other places in this file that the indentation is 2 spaces, if you change it back to 2 spaces the diff will get slightly smaller in some places.